### ISSUE TITLE
`--main-path` option not recognized in Nuitka 1.6

### ISSUE THREAD

**Issue by beyse** [Nuitka/Nuitka#2277]
After carefully reviewing the [Issue Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md#submitting-an-issue) I allow myself to report a bug with the following information:

<details>
  <summary>Noise, I had to go through</summary>

> Please check whether the bug was already reported or fixed.

I checked and I could not find one.

> Please check out if the develop version of Nuitka works better for you.

No, I want to use a stable version of the software.

> How did you install Nuitka and Python

I used `pip install nuitka` to install nuitka using pip.

> Many times when you get an error from Nuitka, your setup may be special

A `print("Hello world")` program does work.

> Also supply a Short, Self Contained, Correct, Example

I considered it and found it not relevant to the problem

> Provide in your issue the Nuitka options used

I state the command line.

> Avoid unnecessary options

Please tell me which options are unnecessary  and I will remove them

> Note if this is a regression

I have no observations on regressions for this project and do not plan to do so.

> Consider getting commercial support

I will consider getting commercial support once I determined it suits all my needs.

</details>

**Description:**
I am attempting to compile multiple Python scripts into a single binary using the `--main-path` option as described in the [Nuitka user manual](https://nuitka.net/doc/user-manual.html#use-case-6-multidist). However, when I run the command, I receive an error stating that `--main-path` is not a recognized option.

**Environment:**
- Nuitka version: 1.6
- Python version: 3.10.11 (main, Apr  5 2023, 14:15:10) [GCC 9.4.0]
- Operating system: Linux, Ubuntu (based on Debian) 20.04.4

**Steps to reproduce:**
1. Create a Makefile with the following content:
    ```Makefile
    # Declare variables
    PYTHON = python
    NUITKA = -m nuitka
    FLAGS = --follow-imports --nofollow-import-to=numpy output-dir=bin
    A_SCRIPT = src/python/apps/script_a.py
    B_SCRIPT = src/python/apps/script_b.py

    # Declare phony targets
    .PHONY: build

    build:
        $(PYTHON) $(NUITKA) $(FLAGS) --main-path=$(A_SCRIPT ) --main-path=$(B_SCRIPT )
    ```

2. Run `make build`.

**Expected outcome:**
Successful compilation of the two Python scripts into a single binary.

**Actual outcome:**
The compilation process fails with the error message: `__main__.py: error: no such option: --main-path`.

**Additional notes:**
From my understanding based on the Nuitka user manual, `--main-path` should be a valid option that can be used to specify multiple main scripts for the compilation process. Any assistance in resolving this issue would be greatly appreciated.

<details>
  <summary>Here are the specific PyPI names and versions</summary>
  

```
aiocache==0.12.0
anyio==3.7.0
arel==0.2.0
astroid==2.15.5
async-asgi-testclient==1.4.11
beautifulsoup4==4.12.2
bidict==0.22.1
blinker==1.6.2
boxx==0.10.0
bpy==3.4.0
certifi==2019.11.28
cffi==1.15.1
chardet==3.0.4
charset-normalizer==3.1.0
click==8.1.3
contourpy==1.0.7
cryptography==41.0.1
cycler==0.11.0
Cython==0.29.35
dacite==1.8.1
dbus-python==1.2.16
debugpy==1.6.7
dill==0.3.6
distinctipy==1.2.2
distro-info===0.23ubuntu1
exceptiongroup==1.1.1
exclusiveprocess==0.9.4
executing==1.2.0
fastapi==0.96.0
ffmpeg-python==0.2.0
Flask==2.3.2
Flask-Cors==3.0.10
fonttools==4.39.4
future==0.18.3
gitdb==4.0.10
GitPython==3.1.31
giturlparse==0.10.0
h11==0.14.0
httptools==0.5.0
idna==2.8
imageio==2.30.0
imutils==0.5.4
iniconfig==2.0.0
isort==5.12.0
itsdangerous==2.1.2
Jinja2==3.1.2
json-merge-patch==0.2
jsonpatch==1.32
jsonpointer==2.3
kiwisolver==1.4.4
lazy-object-proxy==1.9.0
lazy_loader==0.2
loguru==0.7.0
markdown-it-py==2.2.0
MarkupSafe==2.1.3
mathutils==3.3.0
matplotlib==3.7.1
mccabe==0.7.0
mdurl==0.1.2
minexr==1.0.2
multidict==6.0.4
networkx==3.1
Nuitka==1.6
numerize==0.12
numpy==1.24.3
opencv-python==4.5.5.64
ordered-set==4.1.0
packaging==23.1
pandas==1.5.2
Pillow==9.5.0
platformdirs==3.5.1
pluggy==1.0.0
protobuf==3.20.3
psutil==5.9.5
PTable==0.9.2
pycparser==2.21
pydantic==1.10.8
pydicom==2.3.1
Pygments==2.15.1
PyGObject==3.36.0
pylint==2.17.4
pynrrd==0.4.3
PyOpenGL==3.1.7
pyparsing==3.0.9
pytest==7.3.1
python-apt==2.0.1+ubuntu0.20.4.1
python-dateutil==2.8.2
python-dotenv==0.21.1
python-json-logger==2.0.7
python-magic==0.4.27
python-multipart==0.0.5
pytz==2023.3
PyWavelets==1.4.1
PyYAML==6.0
requests==2.31.0
requests-toolbelt==1.0.0
requests-unixsocket==0.2.0
rich==13.4.1
scikit-image==0.18.3
scikit-video==1.1.11
scipy==1.10.1
seaborn==0.12.2
Shapely==1.8.5.post1
SimpleITK==2.2.1
six==1.14.0
smmap==5.0.0
snakeviz==2.2.0
sniffio==1.3.0
soupsieve==2.4.1
starlette==0.27.0
stringcase==1.2.0
supervisely==6.72.28
tifffile==2023.4.12
tomli==2.0.1
tomlkit==0.11.8
tornado==6.3.2
tqdm==4.65.0
trimesh==3.21.7
typing_extensions==4.6.3
tzdata==2023.3
unattended-upgrades==0.1
urllib3==1.26.15
uvicorn==0.22.0
uvloop==0.17.0
varname==0.11.1
watchfiles==0.19.0
watchgod==0.6
websockets==10.4
Werkzeug==2.3.4
wrapt==1.15.0
zstandard==0.21.0
```

</details>

**Comment by kayhayen** [comment]
The docs don't match the code, the option was changed to `--main` at some point during development I guess. Updating the docs to reflect that.

**Comment by beyse** [comment]
> The docs don't match the code, the option was changed to `--main` at some point during development I guess. Updating the docs to reflect that.

Sir, thank you for your timely response. Indeed, using `--main` leads to the expected behavior. 

If you are updating the docs, please note that I also could not find the `--main` nor the `--main-path` option from the help.
Running `python -m nuitka --help | grep -- '--main'` returns no results
 
![image](https://github.com/Nuitka/Nuitka/assets/5306790/1356bba8-ae79-48a7-a5f2-771cf25a656c)

**Comment by kayhayen** [comment]
I guess `help=SUPPRESS_HELP` does not help it being there, that doesn't mean it doesn't exist though. I have to admit, not making it easy to use.

**Comment by kayhayen** [comment]
Going to add help. I guess I was too scared to even attempt to describe what it is, because it is orthogonal to standalone/onefile/accelerated mode. The multidist name is more a goal, namely standalone of multiple programs, and not exactly what it does.

But alas, adding the help output now.

**Comment by kayhayen** [comment]
Yes, that's true. Windows symlinks, CMD files, shell scripts, posix symlinks, they all could be created indeed. But maybe that's for somebody else to PR into Nuitka. It's not for lack of ideas that this is not done. :-)

**Comment by beyse** [comment]
Fair.

So from my point of view the issue is resolved but I leave it up to you to close it, once everything is updated.

In case someone in the future runs into the same problem for some reason, I am going to leave this here:

# ✨ SOLUTION ✨

Works for 1.6

Use `--main` instead of `--main-path`

**Comment by kayhayen** [comment]
So, the doc updates will be added to 1.6.2 and this will get a factory tag once it has a fix, and will be closed, once the 1.6.2 release is done. Which may not be off that far in the distance.

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.6.2 that I just made.
